libraries/chain/controller.cpp

```diff
@@ -4267,81 +4260,13 @@ struct controller_impl {
-         if constexpr (std::is_same_v<BSP, typename std::decay_t<decltype(forkdb.root())>>) {
-            forkdb.add( bsp, ignore_duplicate_t::yes );
-         } else {
-            EOS_ASSERT( false, unlinkable_block_exception,
-                        "wrong block state type, unlinkable block ${id} previous ${p}", ("id", bsp->id())("p", bsp->previous()) );
-         }
-
-         emit( accepted_block_header, std::tie(bsp->block, bsp->id()), __FILE__, __LINE__ );
-      };
-
-      fork_db.apply<void>(do_accept_block);
-   }
-
-   template <class BSP>
-   void push_block( controller::block_report& br,
-                    const BSP& bsp,
-                    const forked_callback_t& forked_branch_cb,
-                    const trx_meta_cache_lookup& trx_lookup )
-   {
-      assert(bsp && bsp->block);
-
-      controller::block_status s = controller::block_status::complete;
-      EOS_ASSERT(!pending, block_validate_exception, "it is not valid to push a block when there is a pending block");
-
-      auto reset_prod_light_validation = fc::make_scoped_exit([old_value=trusted_producer_light_validation, this]() {
-         trusted_producer_light_validation = old_value;
-      });
-      try {
-         const auto& b = bsp->block;
-
-         auto do_push = [&](auto& forkdb) {
-            if constexpr (std::is_same_v<BSP, typename std::decay_t<decltype(forkdb.root())>>) {
-               forkdb.add( bsp, ignore_duplicate_t::yes );
-            } else {
-               EOS_ASSERT( false, unlinkable_block_exception,
-                           "unexpected block state type, unlinkable block ${id} previous ${p}", ("id", bsp->id())("p", b->previous) );
-            }
-
-            if (is_trusted_producer(b->producer)) {
-               trusted_producer_light_validation = true;
-            };
-
-            emit( accepted_block_header, std::tie(bsp->block, bsp->id()), __FILE__, __LINE__ );
-
-            if( read_mode != db_read_mode::IRREVERSIBLE ) {
-               if constexpr (std::is_same_v<BSP, typename std::decay_t<decltype(forkdb.root())>>)
-                  maybe_switch_forks( br, forkdb.head(include_root_t::yes), s, forked_branch_cb, trx_lookup );
-            } else {
-               log_irreversible();
-               transition_to_savanna_if_needed();
-            }
-         };
-
-         fork_db.apply<void>(do_push);
-
-      } FC_LOG_AND_RETHROW( )
-   }
-
-   template <class BSP>
-   void replay_push_block( const signed_block_ptr& b, controller::block_status s ) {
+   void replay_irreversible_block( const signed_block_ptr& b ) {
       validate_db_available_size();
 
       EOS_ASSERT(!pending, block_validate_exception, "it is not valid to push a block when there is a pending block");
 
       try {
          EOS_ASSERT( b, block_validate_exception, "trying to push empty block" );
-         EOS_ASSERT( (s == controller::block_status::irreversible || s == controller::block_status::validated),
```

Comment:
Should this be replaced with a small assert?

Reply:
There is no `s` to assert anymore.